geo/src/algorithm/sweep/mod.rs

```diff
@@ -1,52 +1,78 @@
+//    and adds them as new events. This implementation instead relies on iteration, and yields
+//    intersections immediately as they are discovered.
+//
+// ### Algorithm Flow
+//
+// 1. **Preprocessing**:
+//    - Convert line segments to `SweepLineInterval`s with normalized x-coordinates
+//    - Sort intervals by their left (INSERT) x-coordinate
+//
+// 2. **Sweep Execution**:
+//    - Process intervals in sorted order (left to right).
+//    - For each interval, check all subsequent intervals that start before it ends. These intervals overlap in their x-coordinates and so are candidates for intersection.
+//    - Apply `line_intersection` test to confirm actual intersections.
+//
+// 3. **Output**:
+//    - Yield intersection results as they are discovered during the sweep.
+//
+// This approach maintains the O((n + k) log n) complexity of the classical BO algorithm
+// with fewer concepts and data structures, and allows for lazy iteration of intersections.
+//
+// [Bentley-Ottmann]: https://en.wikipedia.org/wiki/Bentley%E2%80%93Ottmann_algorithm
+// [Chen & Chan, 2003]: https://en.wikipedia.org/wiki/Bentley%E2%80%93Ottmann_algorithm#CITEREFChenChan2003
 
 use crate::line_intersection::{line_intersection, LineIntersection};
 use crate::{GeoFloat, Line};
-use std::cmp::Ordering;
 
 #[cfg(test)]
 mod tests;
 
-/// Iterator over all intersections of a collection of [`Line`]s.
+/// Find all intersections within a collection of [`Line`]s using a simplified Bentley-Ottmann
+/// sweep.
+///
+/// Yields tuples `(`Line`, `Line`, `LineIntersection`)` for each pair of input lines that
+/// intersect.
 ///
-/// # Performance Note
+/// This is a drop-in replacement for computing [`LineIntersection`] over all pairs of the
+/// collection.
 ///
-/// This implementation is most useful when there is a need to efficiently find sparse intersections
-/// between many line segments. For smaller numbers of segments, a brute-force approach
-/// will be around 20 % faster in current tests: if you are reasonably confident that the number of
-/// segments is larger than the number of intersections, use this algorithm.
+/// ## Performance Characteristics
 ///
-/// Yields tuples `(`Line`, `Line`, `LineIntersection`)` for each pair of input
-/// lines that intersect or overlap. This is a drop-in
-/// replacement for computing [`LineIntersection`] over all pairs of
-/// the collection, with the perf caveat noted above.
+/// This implementation is most effective when you have many line segments with sparse intersections,
+/// but even quite dense intersections are competitive in all but extreme cases.
 ///
-/// The implementation uses the [Bentley-Ottmann] sweep line algorithm,
-/// which runs in `O((n + k) log n)` time, where `n` is the number of line segments
-/// and `k` is the number of intersections.
+/// As a rule of thumb, if each segment intersects fewer than 10% of the other segments, you can be
```

Comment:
NBD, but `total` is more intuitive to me than `other` (unless you avoided `total` because it's not strictly accurate)
```suggestion
/// As a rule of thumb, if each segment intersects fewer than 10% of the total segments, you can be
```